libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/Gpg45ProfileEvaluator.java

```diff
@@ -202,15 +203,17 @@ && doesEvidenceContainEvidenceType(evidenceItem, evidenceType)) {
 
     public int calculateF2FRequiredStrengthScore(List<RequiredGpg45ScoresDto> requiredGpg45Scores) {
         int strengthScore = 0;
-        for (RequiredGpg45ScoresDto gpg45Score : requiredGpg45Scores) {
-            Gpg45ScoresDto requiredScores = gpg45Score.getRequiredScores();
-            if (requiredScores.getFraud() > 0 || requiredScores.getActivity() > 0) {
-                continue;
-            }
-            List<EvidenceDto> evidences = requiredScores.getEvidences();
-            if (!evidences.isEmpty()
-                    && (strengthScore == 0 || evidences.get(0).getStrength() < strengthScore)) {
-                strengthScore = evidences.get(0).getStrength();
+        if (!CollectionUtils.isNullOrEmpty(requiredGpg45Scores)) {
```

Comment:
This seems a bit suspicious - when will this be null or empty?

Reply:
During test come across this NPE ...so applied this change.

Reply:
Ok - do you know how that happened? If we're finding that the required scores are `null`, it suggests we're not calculating the diff as expected. We should generate and store this diff in the evaluate GPG45 lambda, but maybe we're skipping that step because of the CI (i.e. it's failing early from CI scoring)...

In any case, I don't think we should be sending a strength score of 0 to F2F - that's not right.

@MikeCollingwood  - this might be even stronger motivation to move the calculation here. It looks like we're skipping the evaluate GPG45 step when on a mitigation journey.

Reply:
It might be that we want to leave this in for now, but then I'd like a link to @MikeCollingwood 's ticket - PYIC-3753, so we don't forget.

e.g.
```
TODO: PYIC-3753 This check is temporarily required until we move the calculation here
```

Reply:
Now https://govukverify.atlassian.net/browse/PYIC-3779